DTML.EduBot/Dialogs/ChitChatDialog.cs

```diff
@@ -3,24 +3,62 @@
     using System;
     using System.Threading.Tasks;
     using Microsoft.Bot.Builder.Dialogs;
-    using Microsoft.Bot.Builder.Luis;
     using Microsoft.Bot.Builder.Luis.Models;
+    using Microsoft.Bot.Connector;
     using DTML.EduBot.Common;
-    using DTML.EduBot.Qna;
-    using Microsoft.Azure;
+    using DTML.EduBot.UserData;
     using Attributes;
 
     [PreConfiguredLuisModel]
     [PreConfiguredQnaModel]
     [Serializable]
     public partial class ChitChatDialog : QnaLuisDialog<object>
     {
+        private readonly LevelDialog _levelDialog;
+        private readonly IUserDataRepository _iUserDataRepository;
+
+        public ChitChatDialog(LevelDialog levelDialog, IUserDataRepository iUserDataRepository)
+        {
+            _levelDialog = levelDialog;
+            _iUserDataRepository = iUserDataRepository;
+        }
+
         [LuisIntent("")]
         [LuisIntent("None")]
         public async Task HandleUnrecognizedIntent(IDialogContext context, LuisResult result)
         {
-            await context.PostAsync(BotPersonality.BotResponseUnrecognizedIntent);
+            string translatedBotResponse = await this.TranslateBotResponseAsync(context, BotPersonality.BotResponseUnrecognizedIntent);
+            await context.PostAsync(translatedBotResponse);
+        }
+       
+        protected override async Task MessageReceived(IDialogContext context, IAwaitable<IMessageActivity> item)
+        {
+            if ((await item) != null && (await item).Text != null)
```

Comment:
var activity = await item instead of "await item" multiple times. 

I am not very confident awaiting a task multiple times.